packages/react-combobox/src/components/Option/useOption.ts

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+import { useContextSelector } from '@fluentui/react-context-selector';
+import { ListboxContext } from '../../contexts/ListboxContext';
+import type { OptionProps, OptionState } from './Option.types';
+
+// TODO: refine this more
+function getValueString(value: string | undefined, children: React.ReactNode) {
+  if (value) {
+    return value;
+  }
+
+  let valueString = '';
+  React.Children.map(children, child => {
+    if (typeof child === 'string') {
+      valueString += child;
+    }
+  });
+  return valueString;
+}
+
+/**
+ * Create the state required to render Option.
+ *
+ * The returned state can be modified with hooks such as useOptionStyles,
+ * before being passed to renderOption.
+ *
+ * @param props - props from this instance of Option
+ * @param ref - reference to root HTMLElement of Option
+ */
+export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElement>): OptionState => {
+  const { activeOption, idBase, onOptionClick, registerOption, selectedKeys, unRegisterOption } = useContextSelector(
+    ListboxContext,
+    ctx => ({
+      activeOption: ctx.activeOption,
+      idBase: ctx.idBase,
+      onOptionClick: ctx.onOptionClick,
+      registerOption: ctx.registerOption,
+      selectedKeys: ctx.selectedKeys,
+      unRegisterOption: ctx.unRegisterOption,
+    }),
+  );
+  const { id = `${idBase}-${props.fluentKey}`, fluentKey: key, disabled, value } = props;
```

Comment:
Instead of using `id`, `key`s or something better I suggest to make `value` a required prop.

- `value` should be unique
- `value` is [an attribute on `<option>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option)
- `value` is a string

Reply:
> An example of this is a combobox that has several pinned options at the top whose values are the same as the unpinned options in the full list (seen this in a Microsoft product in the past).

Do you I correctly understand that it's something like that (_a snippet below_)?

```tsx
function App() {
  return (
    <Combobox>
      <PinnedOptions>
        <Option value="us">US</Option>
        <Option value="canada">Canada</Option>
      </PinnedOptions>
      <AllOptions>
        <Option value="us">US</Option>
        <Option value="canada">Canada</Option>
        <Option value="cz">Czech Republic</Option>
        <Option value="de">Germany</Option>
      </AllOptions>
    </Combobox>
  );
}
```

- React `key`s are not needed there as it's defined in JSX
  - _We don't really need React keys in this snippet_
- Let's assume that `id`s will be autogenerated on every `Option` so they are also uniq
  - They can be used for `aria-activedescendant`
- I expect that values are **not unique** in this scope as if an option is selected in `PinnedOptions` it also should be selected in `AllOptions`

Is there sense in it?

Reply:
Yup, that's essentially the pinned options example I was thinking of.